components/script/dom/readablestreamdefaultreader.rs

```diff
@@ -42,11 +45,22 @@ pub(crate) enum ReadRequest {
     DefaultTee {
         tee_read_request: Dom<DefaultTeeReadRequest>,
     },
+    /// <https://streams.spec.whatwg.org/#read-loop>
+    ReadLoop {
```

Comment:
So this is not necessary, since a normal read request can be used, and some of this data should be moved to either inside the success or failure steps(the promise), or inside either the fulfillment or rejection handler(the bytes for example should go in the fulfillment handler for example). The success and failure steps themselves should be moved to the rejection and fulfillment handers.